src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -261,280 +212,340 @@ class BasePaymentsPage extends React.Component {
-                    </OfflineWithFeedback>
-                </View>
-                <AddPaymentMethodMenu
-                    isVisible={this.state.shouldShowAddPaymentMenu}
-                    onClose={this.hideAddPaymentMenu}
-                    anchorPosition={{
-                        horizontal: this.state.anchorPositionHorizontal,
-                        vertical: this.state.anchorPositionVertical - 10,
-                    }}
-                    onItemSelected={(method) => this.addPaymentMethodTypePressed(method)}
-                />
-                <Popover
-                    isVisible={this.state.shouldShowDefaultDeleteMenu}
-                    onClose={this.hideDefaultDeleteMenu}
-                    anchorPosition={{
-                        top: this.state.anchorPositionTop,
-                        right: this.state.anchorPositionRight,
-                    }}
+                    </View>
+                    {props.userWallet.currentBalance > 0 && (
+                        <View style={styles.mb3}>
+                            <KYCWall
+                                onSuccessfulKYC={navigateToTransferBalancePage}
+                                enablePaymentsRoute={ROUTES.SETTINGS_ENABLE_PAYMENTS}
+                                addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
+                                addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
+                                popoverPlacement="bottom"
+                            >
+                                {(triggerKYCFlow) => (
+                                    <MenuItem
+                                        title={translate('common.transferBalance')}
+                                        icon={Expensicons.Transfer}
+                                        onPress={triggerKYCFlow}
+                                        shouldShowRightIcon
+                                        disabled={props.network.isOffline}
+                                    />
+                                )}
+                            </KYCWall>
+                        </View>
+                    )}
+                </>
+            )}
+            <Text style={[styles.ph5, styles.textLabelSupporting, styles.mb1]}>{translate('paymentsPage.paymentMethodsTitle')}</Text>
+        </>
+    );
+
+    useEffect(() => {
+        PaymentMethods.openPaymentsPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!props.network.isOffline) {
+            updateShouldShowLoadingSpinner();
+        } else {
+            debounceSetShouldShowLoadingSpinner();
+        }
+    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
+
+    useEffect(() => {
+        if (props.network.isOffline) {
+            return;
+        }
+        PaymentMethods.openPaymentsPage();
+    }, [props.network.isOffline]);
```

Comment:
This can be clubbed together with above

```js
useEffect(() => {
        // If the user was previously offline, skip debouncing showing the loader
        if (!props.network.isOffline) {
            updateShouldShowLoadingSpinner();
            // Here as it is called when reconnecting PaymentMethods.openPaymentsPage();
            
        } else {
            debounceSetShouldShowLoadingSpinner();
        }
    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);

/* Remove this
    useEffect(() => {
        if (props.network.isOffline) {
            return;
        }
        PaymentMethods.openPaymentsPage();
    }, [props.network.isOffline]);*/
```

Reply:
Hi, @Santhosh-Sellavel 
This function `PaymentMethods.openPaymentsPage()` should be called when only network changes but if it will be called infinitely.
Because when `openPaymentsPage` is called, `props.isLoadingPaymentMethods` will be affected, thus useCallback `updateShouldShowLoadingSpinner` will gets updated.
FYI, I tried to merged above 2 useEffect but `openPaymentsPage` was called infinitely and I noticed above reason.

@Santhosh-Sellavel Can you please take a look again?
Thanks.

Reply:
Ah you might be right, I missed that!